server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -720,13 +719,10 @@ private Query getWildcardQuerySingle(String field, String termStr) throws ParseE
         Analyzer oldAnalyzer = getAnalyzer();
         try {
             MappedFieldType currentFieldType = queryBuilder.context.fieldMapper(field);
-            if (currentFieldType == null) {
-                return newUnmappedFieldQuery(field);
-            }
-            if (forceAnalyzer != null && (analyzeWildcard || currentFieldType.getTextSearchInfo().isTokenized())) {
-                setAnalyzer(forceAnalyzer);
-                return super.getWildcardQuery(currentFieldType.name(), termStr);
+            if (currentFieldType != null) {
```

Comment:
Hm ... did we lost `if (currentFieldType == null) {` condition?

Reply:
I figured it will be cleaner to check `if (currentFieldType != null)` now that we don't use that we always set the analyzer

Reply:
But this is unmapped field case, we probably should return `newUnmappedFieldQuery` ?